frontend/src/libs/components/article-form/article-form.tsx

```diff
@@ -0,0 +1,106 @@
+import { ButtonType } from '~/libs/enums/enums.js';
+import { useAppDispatch, useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import {
+  articleCreateValidationSchema,
+  type ArticleRequestDto,
+} from '~/packages/articles/articles.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
+import { Button, Input, Textarea } from '../components.js';
+import { DEFAULT_ARTICLE_FORM_PAYLOAD } from './libs/constants.js';
+import { ArticleSubmitType } from './libs/enums/enums.js';
+import styles from './styles.module.scss';
+
+// TODO: Should be moved out from common components.
+const ArticleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { control, errors, handleSubmit, reset, dirtyFields } =
+    useAppForm<ArticleRequestDto>({
+      defaultValues: DEFAULT_ARTICLE_FORM_PAYLOAD,
+      validationSchema: articleCreateValidationSchema,
+    });
+
+  const areFieldsDirty = !!dirtyFields.title && !!dirtyFields.text;
```

Comment:
Is there any considerations not to use isDirty from useForm instead of manually check each dirtyField?

Reply:
I tried to use `disabled={!isDirty}` on the “Save Draft” and “Publish” buttons, but in this case, they become available if any of the fields are not empty. But they should be enabled only if both title and text are not empty. Maybe there is an approach with isDirty I missed?

Reply:
isDirty is for whether the fields have changed. And I used it to disable the save button in edit user info

Reply:
Let's handle it with isDirty per whole form, and mark this discussion in comment to ticket (to make it clear with acceptance criterias)

Reply:
Yep